### PR TITLE
CORE-9379 Verification Sandbox accepts only contract CPKs

### PR DESCRIPTION
**Pull request by mbrkic-r3**
Using `CpkReadService` to retrieve CPK metadata and check whether CPKs for Verification Sandbox are contract CPKs.

### CODE REVIEW

**Review by lankydan — DISMISSED** [review]

File: components/ledger/ledger-verification/src/main/kotlin/net/corda/ledger/verification/sandbox/impl/VerificationSandboxServiceImpl.kt [hunk 1]
```diff
@@ -28,51 +29,40 @@ import org.slf4j.LoggerFactory
...
-            throw NotAllowedCpkException(
-                "CPKs for Verification Sandbox for $holdingIdentity are not allowed: $cpkIds"
+    private fun checkCpks(holdingIdentity: HoldingIdentity, cpks: List<CordaPackageSummary>) {
+        cpks.forEach {
+            val cpk = cpkReadService.get(it.fileChecksum.toSecureHash()) ?: throw CpkNotAvailableException(
+                "CPK for Verification Sandbox for $holdingIdentity is not available (yet): $it"
             )
+            if (!cpk.metadata.isContractCpk()) {
+                throw NotAllowedCpkException(
+                    "CPK for Verification Sandbox for $holdingIdentity is not allowed: $it"
```
Comment (blsemo) [inline components/ledger/ledger-verification/src/main/kotlin/net/corda/ledger/verification/sandbox/impl/VerificationSandboxServiceImpl.kt]: I don't like this error message, it's  not very clear. How about something like this:
```suggestion
                    "This CPK is not allowed for use in the Verification Sandbox for $holdingIdentity - only contract CPKs are allowed: $it"
```

File: components/ledger/ledger-verification/src/main/kotlin/net/corda/ledger/verification/sandbox/impl/VerificationSandboxServiceImpl.kt [hunk 2]
```diff
@@ -28,51 +29,40 @@ import org.slf4j.LoggerFactory
...
-            .filterNot { it.isContractCpk() }
-
-        if (nonContractCpks.isNotEmpty()) {
-            val cpkIds = nonContractCpks.map { it.cpkId }
-            throw NotAllowedCpkException(
-                "CPKs for Verification Sandbox for $holdingIdentity are not allowed: $cpkIds"
+    private fun checkCpks(holdingIdentity: HoldingIdentity, cpks: List<CordaPackageSummary>) {
+        cpks.forEach {
+            val cpk = cpkReadService.get(it.fileChecksum.toSecureHash()) ?: throw CpkNotAvailableException(
+                "CPK for Verification Sandbox for $holdingIdentity is not available (yet): $it"
```
Comment (blsemo) [inline components/ledger/ledger-verification/src/main/kotlin/net/corda/ledger/verification/sandbox/impl/VerificationSandboxServiceImpl.kt]:
```suggestion
                "This CPK is not available (yet) for the Verification Sandbox for $holdingIdentity: $it"
```

**Review by blsemo — COMMENTED** [review]
Looks good, I just would like to make the error messages easier to understand.

File: components/ledger/ledger-verification/src/main/kotlin/net/corda/ledger/verification/sandbox/impl/VerificationSandboxServiceImpl.kt (hunk 2 above)
Comment (mbrkic-r3) [reply components/ledger/ledger-verification/src/main/kotlin/net/corda/ledger/verification/sandbox/impl/VerificationSandboxServiceImpl.kt]: Committed suggested changes, please re-approve

**Review by ronanbrowne — APPROVED** [review]
LGTM (only build reviewed)

**Review by blsemo — APPROVED** [review]
LGTM

**Pull request merged**